### PR TITLE
Declarative apis

### PR DESCRIPTION
**Pull request by DanRoseus**
- Use Product in EEI/DDD pipelines to publish to APIC
- Add script to support subscribing to a product and creating a secret with credentials, both client id and also client secret
- Change the IntegrationRuntimes to remove old configurations and instead use CR settings for:
  - Forcing https rather than http
  - Disabling use of routes, instead using a service
  - Setting up/using basic auth
- Change test scripts for DDD to pass client secret when calling APIC and to get ACE creds from the relevant secret
- Update EEI docs/example yaml for 11.2

DDD pipelines all run:
image

Able to GET/POST for EEI:
```
fermi-dan-1-click-dev $ curl -k -H "X-IBM-Client-Id: ${API_CLIENT_ID}" "${API_BASE_URL}/quote?QuoteID=$QUOTE_ID"
{
    "address": "790 Arrowhead Court, Portsmouth",
    "licensePlate": "WMC-9628",
    "name": "Ronny Doyle",
    "descriptionOfDamage": "Wheel fell off",
    "usState": "VA",
    "quoteid": "01fe483a-c422-45c0-801e-930dede61f8e",
    "email": "RonnyDoyle@mail.com",
    "claimStatus": "7"
}

fermi-dan-1-click-dev $ curl -k "${API_BASE_URL}/quote" \
>   -H "X-IBM-Client-Id: ${API_CLIENT_ID}" \
>   -d '{
>     "name": "Barack Obama",
>     "email": "comments@whitehouse.gov",
>     "age": "50",
>     "address": "1600 Pennsylvania Avenue",
>     "usState": "DC",
>     "licensePlate": "EK 3333",
>     "descriptionOfDamage": "420"
>   }'
{"name":"Barack Obama","email":"comments@whitehouse.gov","age":"50","address":"1600 Pennsylvania Avenue","usState":"DC","licensePlate":"EK 3333","descriptionOfDamage":"420","quoteid":"9244d2e2-ff89-11ed-99f5-ac1e4fe90000"}
```

### CODE REVIEW

File: products/bash/apic-subscribe.sh [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+if [ -z "${CONSUMER_ORG}" ]; then CONSUMER_ORG=${PROVIDER_ORG}-corp; fi
+
+# If APP not set then default based on PRODUCT
+if [ -z "${APP}" ]; then APP=${PRODUCT}-app; fi
+
+TICK="\xE2\x9C\x85"
+CROSS="\xE2\x9D\x8C"
+
+client_id="599b7aef-8841-4ee2-88a0-84d49c4d6ff2"
+client_secret="0ea28423-e73b-47d4-b40e-ddb45c48bb0c"
```
Comment (laurensks) [inline products/bash/apic-subscribe.sh:74]: is the client id and secret the same each time? I would have assumed that a id would change everytime it is created?

File: products/bash/apic-subscribe.sh [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+RES=$(curl -kLsS https://$PLATFORM_API_EP/api/catalogs/${PROVIDER_ORG}/$CATALOG/configured-catalog-user-registries \
+  -H "accept: application/json" \
+  -H "authorization: Bearer ${TOKEN}")
+handle_res "${RES}"
+
+USER_REGISTRY_URL=$(echo "${OUTPUT}" | $JQ -r ".results[0].user_registry_url")
+$DEBUG && echo "[DEBUG] User registry url: ${USER_REGISTRY_URL}"
+echo -e "[INFO] ${TICK} Got configured catalog user registry url for ${PROVIDER_ORG}-catalog"
+
+$DEBUG && echo "[DEBUG] username: $CORG_OWNER_USERNAME"
```
Comment (laurensks) [inline products/bash/apic-subscribe.sh:139]:
```suggestion
$DEBUG && echo "[DEBUG] username: ${CORG_OWNER_USERNAME}"
```

File: products/bash/apic-subscribe.sh (hunk 1 above)
Comment (DanRoseus) [reply products/bash/apic-subscribe.sh:74]: This is the standard client_id/client_secret that is used to interact with the APIM REST API and it doesn't change. Note it is not the same as the client_id/client_secret that are different for every application that is used to call the DDD/EEI endpoints and is extracted on lines 200/201.

File: products/bash/install-ocp-pipeline.sh [hunk 3]
```diff
@@ -24,7 +24,7 @@ metadata:
   name: openshift-pipelines-operator
   namespace: openshift-operators
 spec:
-  channel: stable
+  # channel: stable
```
Comment (laurensks) [inline products/bash/install-ocp-pipeline.sh:27]: is there a reason for the channel to be commented out?

File: products/bash/apic-subscribe.sh (hunk 2 above)
Comment (DanRoseus) [reply products/bash/apic-subscribe.sh:139]: Doesn't really seem a necessary change, it works fine with the other syntax. I guess you just prefer the `{}` version?

File: products/bash/install-ocp-pipeline.sh (hunk 3 above)
Comment (DanRoseus) [reply products/bash/install-ocp-pipeline.sh:27]: With it commented out it used the default channel, which is usually the latest. I'm not seeing it on my current cluster, but on a previous cluster I think the stable channel didn't exist. Not sure if that was an OCP 4.12 thing. I think the change is safe, and we can re-test as part of the 2023.2.1 update, when we'll need to test on both 4.10 and 4.12

File: products/bash/apic-subscribe.sh (hunk 2 above)
Comment (laurensks) [reply products/bash/apic-subscribe.sh:139]: just would have matched with the other debug but its fine :D

File: products/bash/install-ocp-pipeline.sh (hunk 3 above)
Comment (laurensks) [reply products/bash/install-ocp-pipeline.sh:27]: Good point alright we'll leave it in then :)

File: products/bash/apic-subscribe.sh (hunk 1 above)
Comment (laurensks) [reply products/bash/apic-subscribe.sh:74]: got you 👍

**Review by laurensks — APPROVED** [review]
Looks good :)

**Pull request merged**